libcontainer/rootfs_linux.go

```diff
@@ -232,6 +236,27 @@ func prepareRootfs(pipe *syncSocket, iConfig *initConfig) (err error) {
 	return nil
 }
 
+// adjustRootMountPropagation applies mount propagation flags such as MS_SHARED or MS_UNBINDABLE
+// to the root mount after the pivot_root or chroot has taken place.
```

Comment:
Needs a comment to explain why this cannot be done in `prepareRoot`:
https://github.com/opencontainers/runc/blob/3d8a278bdd7d76915878ef6bdd0e5f031c589f30/libcontainer/rootfs_linux.go#L1012-L1027

Reply:
I have added a comment.